File: src/cpp_pkg/src/minimal_node.cpp
#include "rclcpp/rclcpp.hpp"

class MinimalNode : public rclcpp::Node
{
    public:
        MinimalNode() : Node("minimal_node")
        {
            timer_ = this->create_wall_timer(std::chrono::seconds(1), std::bind(&MinimalNode::timer_callback, this));
        }
        
    private:
        void timer_callback()
        {
            RCLCPP_INFO(this->get_logger(), "Minimal node is running ...");
        }

        rclcpp::TimerBase::SharedPtr timer_;
}; 

int main(int argc, char **argv){

    rclcpp::init(argc, argv);
    auto node = std::make_shared<MinimalNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/py_pkg/py_pkg/minimal_node.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node


class MininalNode(Node):

    def __init__(self):
        super().__init__("minimal_node")
        self.create_timer(1.0, self.timer_callback)

    def timer_callback(self):
        self.get_logger().info("Minimal node is running ...")

def main(args=None):
    rclpy.init(args=args)
    node = MininalNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()